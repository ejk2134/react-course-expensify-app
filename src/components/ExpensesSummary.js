import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getExpensesTotal from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => (
    <div>
        <p>Viewing {expenseCount} {expenseCount === 1 ? 'item' : 'items'} totalling {expensesTotal}</p>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenseCount: getVisibleExpenses(state.expenses, state.filters).length,
        expensesTotal: numeral(getExpensesTotal(getVisibleExpenses(state.expenses, state.filters)) / 100).format('$0,0.00')
    };
};

export default connect(mapStateToProps)(ExpensesSummary);