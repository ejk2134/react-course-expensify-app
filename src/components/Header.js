import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>home </NavLink>
        <NavLink to="/create" activeClassName="is-active">new </NavLink>
    </header>
);

export default Header;