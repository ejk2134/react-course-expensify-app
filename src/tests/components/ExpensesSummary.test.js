import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={'$5.00'}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with more than one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={'$5.00'}/>);
    expect(wrapper).toMatchSnapshot();
});