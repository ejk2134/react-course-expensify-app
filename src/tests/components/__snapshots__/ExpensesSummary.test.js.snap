// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with more than one expense 1`] = `
<div>
  <p>
    Viewing 
    2
     
    items
     totalling 
    $5.00
  </p>
</div>
`;

exports[`should render ExpensesSummary with one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    item
     totalling 
    $5.00
  </p>
</div>
`;
